### PR TITLE
Optimizer+

### PR DESCRIPTION
**Pull request by nicktheracker**

### CODE REVIEW

**Review by leo-bertini — APPROVED** [review]

File: optimizer-plus/optimizer-plus-sd-policy/base-policy.tf [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+}
+
+resource "google_monitoring_alert_policy" "disk_usage" {
+  display_name = "rax-optimizer-plus-monitoring-disk_usage"
+  combiner     = "AND_WITH_MATCHING_RESOURCE"
+  enabled      = true  
+  conditions {
+    display_name = "Metric Threshold on All Instance (GCE)s"
+    condition_threshold {
+      filter     = <<EOT
```
Comment (rs-danielg) [inline optimizer-plus/optimizer-plus-sd-policy/base-policy.tf:101]: Needs to ignore various devices, loop is for snaps which are always 100%, tmpfs/udev are memory systems.
`metric.label."device"!=monitoring.regex.full_match(".*(loop[0-9]|tmpfs|udev).*")`

File: optimizer-plus/optimizer-plus-sd-policy/variables.tf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    enabled       = false
+    mem_threshold = 90
+  }
+}
+
+variable "disk_usage" {
+  description = "Disk Usage Parameters"
+  type = object({
+    enabled              = bool
+    blk_dev_name         = string
```
Comment (rs-danielg) [inline optimizer-plus/optimizer-plus-sd-policy/variables.tf:40]: blk_dev_name isn't used, it should be removed.

File: optimizer-plus/optimizer-plus-sd-policy/variables.tf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    disk_threshold_percentage = number
+  })
+
+  default = {
+    enabled                   = false
+    blk_dev_name              = "null"
```
Comment (rs-danielg) [inline optimizer-plus/optimizer-plus-sd-policy/variables.tf:46]: blk_dev_name isn't used, it should be removed.

File: optimizer-plus/optimizer-plus-sd-policy/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+```
+module "rs_sd_policy" {
+ source = "git@github.com:racker/mgcp-terraform-modules//optimizer-plus/optimizer-plus-sd-policy/?ref=master"
+
+ enabled = true
+ project_id = "someproject"
+
+disk_usage {
+    enabled = true
+    blk_dev_name = ""
```
Comment (rs-danielg) [inline optimizer-plus/optimizer-plus-sd-policy/README.md:16]: blk_dev_name is not used by monitor, and shouldn't be a variable.

File: optimizer-plus/optimizer-plus-sd-policy/base-policy.tf [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  conditions {
+    display_name = "Uptime check for GCE INSTANCE - Platform"
+    condition_threshold {
+      filter     = <<EOT
+              metric.type="compute.googleapis.com/instance/uptime" AND
+              metadata.user_labels.autoscaled="false" AND
+              metadata.user_labels.monitored="true" AND
+              resource.type="gce_instance"
+      EOT
+      duration   = "900s"
```
Comment (rs-danielg) [inline optimizer-plus/optimizer-plus-sd-policy/base-policy.tf:18]: Could possibly be reduced, 5 minutes is usually the maximum amount of time it takes an instance to go down and come back up. Anything longer is going to be hard down, or boot issues.

**Review by rs-danielg — CHANGES_REQUESTED** [review]

**Pull request merged**